src/relations/db.py

```diff
@@ -168,9 +169,12 @@ def _block_on_extensions(self, relation, remote_app_databag: Dict) -> bool:
 
     def _get_relation_extensions(self, relation: Relation) -> List[str]:
         """Get enabled extensions for a relation."""
-        for data in relation.data.values():
-            if "extensions" in data:
-                return data["extensions"].split(",")
+        try:
+            for data in relation.data.values():
+                if "extensions" in data:
+                    return data["extensions"].split(",")
+        except ModelError:
+            logger.warning(f"Unable to access relation data for relation {relation.id}")
```

Comment:
I've occasionally got a model error here. I think the cause is that the relation data has already been purged by the time we call this.